monai/transforms/post/array.py

```diff
@@ -679,17 +679,21 @@ def __init__(self, weights: Sequence[float] | NdarrayOrTensor | None = None) ->
         self.weights = torch.as_tensor(weights, dtype=torch.float) if weights is not None else None
 
     def __call__(self, img: Sequence[NdarrayOrTensor] | NdarrayOrTensor) -> NdarrayOrTensor:
-        img_ = self.get_stacked_torch(img)
-        if self.weights is not None:
-            self.weights = self.weights.to(img_.device)
-            shape = tuple(self.weights.shape)
-            for _ in range(img_.ndimension() - self.weights.ndimension()):
-                shape += (1,)
-            weights = self.weights.reshape(*shape)
+        out_pt = None
+        total_weight = 0.0
+
+        for i, pred in enumerate(img):
```

Comment:
You may need check the the type for img what if it's a single array. Something like the change here: 
https://github.com/Project-MONAI/MONAI/blob/bae8f59a6d42f0264b083832e543bb671b07a918/monai/transforms/post/array.py#L639-L644